src/cache/mod.rs

```diff
@@ -0,0 +1,225 @@
+// Copyright 2021 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+mod item;
+
+use self::item::Item;
+use std::collections::BTreeMap;
+use std::hash::Hash;
+use std::time::Duration;
+use tokio::sync::RwLock;
+
+///
+#[derive(Debug)]
+pub struct Cache<T, V>
```

Comment:
ideally, this shall be moved into lru_time_cache , maybe adding it as an additional facility.
btw. just wondering is it possible to make LruCache to be RwLock ? 
so that this Cache becomes a RwLock warrper of LruCache ?

Reply:
Will be addressing all cache things (still wip) in follow up PR.